### ISSUE TITLE
Terrain textures disappear

### ISSUE THREAD

**Issue by SomenameIchose** [pioneerspacesim/pioneer#5806]
### Observed behaviour
When approaching the planet at a certain distance some terrain textures disappear.
On a first screenshot textures are proper
![screenshot-20240324-215601](https://github.com/pioneerspacesim/pioneer/assets/154741968/3a81abda-3c1b-4a00-8691-815ab9e8216f)
But at some point they(textures) start bugging like this
![screenshot-20240324-215625](https://github.com/pioneerspacesim/pioneer/assets/154741968/434d3f3d-a077-4f69-8317-b3c552ecb0a4)
![screenshot-20240324-215536](https://github.com/pioneerspacesim/pioneer/assets/154741968/c15eb0d6-2f66-4603-ac0e-d4c59d82f741)
![screenshot-20240324-215524](https://github.com/pioneerspacesim/pioneer/assets/154741968/5259e6e8-6304-4518-8460-9194e1c29443)

### Expected behaviour
Terrain textures should be stable and shouldn't disappear

### Steps to reproduce
Fly up to critical distance to certain planets and moons (in this case it is Pallas, I also saw this bug approaching Thebe)

__My pioneer version (and OS):__
ver 20240314 (8b6ae92)
OS: Windows 10 Home Edition

__My output.txt (required) and game save (optional, but recommended)__
[output.txt](https://github.com/pioneerspacesim/pioneer/files/14736373/output.txt)
[save.zip](https://github.com/pioneerspacesim/pioneer/files/14736377/save.zip)

**Comment by fluffyfreak** [comment]
That's not rendering the patch at all it seems, almost like the culling has decided it isn't visible

**Comment by fluffyfreak** [comment]
I've not been able to repro' this with some quick local testing around Pallas or Thebe. I also tried with the supplied savegame.

@SomenameIchose what GPU do you have?

**Comment by SomenameIchose** [comment]
> I've not been able to repro' this with some quick local testing around Pallas or Thebe. I also tried with the supplied savegame.
> 
> @SomenameIchose what GPU do you have?
Intel(R) Pentium(R) Gold G5420 CPU @ 3.80GHz   3.79 GHz

**Comment by SomenameIchose** [comment]
I gave an another try to fly to Pallas but textures don't disappear this time. Maybe it happens at certain coordinates?

**Comment by SomenameIchose** [comment]
Yes, I tested they disappear at coordinates of S029°25'12 E042°42'07

**Comment by pcercuei** [comment]
I can reproduce, see the attached save file. Just load it in Pioneer and you should see it.

[Jameson2.zip](https://github.com/pioneerspacesim/pioneer/files/14962854/Jameson2.zip)

**Comment by pcercuei** [comment]
Note that it happens both on my integrated Intel card and my discrete Nvidia card (with Nouveau).

**Comment by fluffyfreak** [comment]
This isn't a GPU/driver bug, this is CPU side terrain generation and the rendering calls itself. Can confirm that it's CPU side only so you'll see it on anything that can run it.

**Comment by fluffyfreak** [comment]
I can confirm that it's the Horizon Culling optimisation which is... over optimising for some reason 🤷 however nothing appears to have changed in the `SSphere::HorizonCulling` method in years. The only change I can see is to the centroid in GeoPatch but I don't know how that might be affecting it.

**Comment by fluffyfreak** [comment]
Oof nope that ain't it either, you have found me a real doozy of a bug 🤣

**Comment by fluffyfreak** [comment]
Ok, I might have a fix for this in the works but it's after midnight, so I'll continue another day

**Comment by fluffyfreak** [comment]
I've put in a first draft of a fix in #5829 which will need some serious testing.
